dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/CurrentUserUtil.java

```diff
@@ -173,20 +176,96 @@ public static <T> T getUserSetting( UserSettingKey key )
+        initializeAndUnproxy( user.getDataViewOrganisationUnits() );
+        user.getDataViewOrganisationUnits().stream().filter( Objects::nonNull )
+            .forEach( organisationUnit -> initializeAndUnproxy( organisationUnit.getChildren() ) );
+
         initializeAndUnproxy( user.getUserRoles() );
         for ( UserRole userRole : user.getUserRoles() )
         {
+            if ( userRole == null )
+            {
+                continue;
+            }
             initializeAndUnproxy( userRole );
             initializeAndUnproxy( userRole.getAuthorities() );
+            initializeAndUnproxy( userRole.getMembers() );
+            initializeAndUnproxy( userRole.getUserAccesses() );
+            initializeAndUnproxy( userRole.getAttributeValues() );
         }
+
         initializeAndUnproxy( user.getGroups() );
-        initializeAndUnproxy( user.getTeiSearchOrganisationUnits() );
-        initializeAndUnproxy( user.getDataViewOrganisationUnits() );
+        for ( UserGroup group : user.getGroups() )
+        {
+            if ( group == null )
+            {
+                continue;
+            }
+            initializeAndUnproxy( group.getManagedByGroups() );
+            initializeAndUnproxy( group.getManagedGroups() );
+            initializeAndUnproxy( group.getMembers() );
+            initializeAndUnproxy( group.getAttributeValues() );
+            initializeAndUnproxy( group.getTranslations() );
+            initializeAndUnproxy( group.getUserAccesses() );
+            initializeAndUnproxy( group.getUserGroupAccesses() );
+        }
+
         initializeAndUnproxy( user.getCogsDimensionConstraints() );
+        for ( CategoryOptionGroupSet groupSet : user.getCogsDimensionConstraints() )
+        {
+            if ( groupSet == null )
+            {
+                continue;
+            }
+            initializeAndUnproxy( groupSet.getMembers() );
+            initializeAndUnproxy( groupSet.getItems() );
+            initializeAndUnproxy( groupSet.getFilterItemsAsList() );
+            initializeAndUnproxy( groupSet.getAttributeValues() );
+            initializeAndUnproxy( groupSet.getFavorites() );
+            initializeAndUnproxy( groupSet.getUserAccesses() );
+            initializeAndUnproxy( groupSet.getUserGroupAccesses() );
+            initializeAndUnproxy( groupSet.getTranslations() );
+
+        }
+
         initializeAndUnproxy( user.getDimensionConstraints() );
+        for ( DimensionalObject dimension : user.getDimensionConstraints() )
+        {
+            if ( dimension == null )
+            {
+                continue;
+            }
+            initializeAndUnproxy( dimension.getItems() );
+            initializeAndUnproxy( dimension.getAttributeValues() );
+            initializeAndUnproxy( dimension.getFavorites() );
+            initializeAndUnproxy( dimension.getUserAccesses() );
+            initializeAndUnproxy( dimension.getUserGroupAccesses() );
```

Comment:
no need to initialize `UserAccesses` and `UserGroupAccesses` here. Those are just DTO collections, mapped from jsonb `dimension.sharing.users` and `dimension.sharing.userGroups`. 

Please note that  each element in collection  `dimension.getUserGroupAccesses()` contains only `uid` and `access` string

```json
 "userGroupAccesses": [
            {
                "id": "hj0nnsVsPLU",
                "access": "rw------"
            },
            {
                "id": "qMjBflJMOfB",
                "access": "r-------"
            }
        ]
```

Reply:
Ok, good point. I will remove those.